src/games/strategy/triplea/ai/proAI/util/ProUtils.java

```diff
@@ -47,24 +47,59 @@ public ProUtils(final ProAI ai) {
     return unitTerritoryMap;
   }
 
-  public List<PlayerID> getEnemyPlayersInTurnOrder(final PlayerID player) {
+  public List<PlayerID> getOtherPlayersInTurnOrder(final PlayerID player) {
     final GameData data = ai.getGameData();
-    final List<PlayerID> enemyPlayers = new ArrayList<PlayerID>();
-    GameSequence sequence = data.getSequence();
-    int startIndex = sequence.getStepIndex();
+    final List<PlayerID> players = new ArrayList<PlayerID>();
+    final GameSequence sequence = data.getSequence();
+    final int startIndex = sequence.getStepIndex();
     for (int i = 0; i < sequence.size(); i++) {
       int currentIndex = startIndex + i;
       if (currentIndex >= sequence.size()) {
         currentIndex -= sequence.size();
       }
-      GameStep step = sequence.getStep(currentIndex);
-      PlayerID stepPlayer = step.getPlayerID();
-      if (step.getName().endsWith("CombatMove") && stepPlayer != null && !enemyPlayers.contains(stepPlayer)
-          && !data.getRelationshipTracker().isAllied(player, stepPlayer)) {
-        enemyPlayers.add(step.getPlayerID());
+      final GameStep step = sequence.getStep(currentIndex);
+      final PlayerID stepPlayer = step.getPlayerID();
+      if (step.getName().endsWith("CombatMove") && stepPlayer != null && !stepPlayer.equals(player)
+          && !players.contains(stepPlayer)) {
+        players.add(step.getPlayerID());
       }
     }
-    return enemyPlayers;
+    return players;
+  }
+
+  public List<PlayerID> getAlliedPlayersInTurnOrder(final PlayerID player) {
+    final GameData data = ai.getGameData();
+    final List<PlayerID> players = getOtherPlayersInTurnOrder(player);
+    for (final Iterator<PlayerID> it = players.iterator(); it.hasNext();) {
+      final PlayerID currentPlayer = it.next();
+      if (!data.getRelationshipTracker().isAllied(player, currentPlayer)) {
+        it.remove();
+      }
+    }
+    return players;
+  }
+
+  public List<PlayerID> getEnemyPlayersInTurnOrder(final PlayerID player) {
```

Comment:
So they are slightly different as one checks allied relationship and one checks the inverse. There is some copy/paste but most of it has already been moved to 'getOtherPlayersInTurnOrder'. There may be a way to take it even further by trying to just pass in the relationship to be tested but this isn't as straightforward since it requires the 'currentPlayer'.